internal/helm/chart/dependency_manager.go

```diff
@@ -35,21 +35,29 @@ import (
 	"github.com/fluxcd/source-controller/internal/helm/repository"
 )
 
-// GetChartRepositoryCallback must return a repository.ChartRepository for the
-// URL, or an error describing why it could not be returned.
-type GetChartRepositoryCallback func(url string) (*repository.ChartRepository, error)
+// CleanDownloader is a Downloader that cleans temporary files created by the downloader,
+// caching the files if the cache is configured,
+// and calling garbage collector to remove unused files.
+type CleanDownloader interface {
```

Comment:
Is this the best name we can come up with? As when I initially read the name, I thought it had to do with "cleanly downloading" the file, and not with garbage collection.

Reply:
another try: `CleanableDownloader`

Reply:
yes I have to agree that with the current state of the code, it would be better. And has it is used in several places, I will move the interface to the repository pkg.